MovieApp/MovieApp/App/Features/Movies/UseCases/MoviesUseCaseProtocol.swift

```diff
@@ -7,10 +7,9 @@ protocol MoviesUseCaseProtocol {
     var oldFavoriteItems: [Int] { get }
 
     func fetchMovies(
-        categoryViewModel: MovieCategoryViewModel,
-        subcategoryViewModel: SubcategoryViewModel,
-        completion: @escaping (Result<[MovieModel], Error>) -> Void
-    )
+        categoryModel: MovieCategoryViewModel,
```

Comment:
This is not correct, UseCase should get *Model classes as an argument, not *ViewModel.